src/main/java/de/tum/in/www1/artemis/service/metis/ConversationMessagingService.java

```diff
@@ -100,20 +110,37 @@ public Post createMessage(Long courseId, Post newMessage) {
 
         var createdMessage = conversationMessageRepository.save(newMessage);
-        broadcastForPost(new PostDTO(createdMessage, MetisCrudAction.CREATE), course);
+
+        // reset the conversation again, because it might have been lost during save
         createdMessage.setConversation(conversation);
+        // reduce the payload of the response / websocket message: this is important to avoid overloading the involved subsystems
+        if (createdMessage.getConversation() != null) {
+            createdMessage.getConversation().hideDetails();
+        }
+
+        // Websocket notification 1: this notifies everyone including the author that there is a new message
+        broadcastForPost(new PostDTO(createdMessage, MetisCrudAction.CREATE), course, notificationRecipients);
+
         if (conversation instanceof OneToOneChat) {
             var getNumberOfPosts = conversationMessageRepository.countByConversationId(conversation.getId());
             if (getNumberOfPosts == 1) { // first message in one to one chat --> notify all participants that a conversation with them has been created
-                var participants = conversationParticipantRepository.findConversationParticipantByConversationId(conversation.getId()).stream()
-                        .map(ConversationParticipant::getUser).filter(Objects::nonNull).collect(Collectors.toSet());
-                conversationService.broadcastOnConversationMembershipChannel(course, MetisCrudAction.CREATE, conversation, participants);
+                // Another websocket notification
+                conversationService.broadcastOnConversationMembershipChannel(course, MetisCrudAction.CREATE, conversation, notificationRecipients);
             }
         }
         conversationParticipantRepository.incrementUnreadMessagesCountOfParticipants(conversation.getId(), author.getId());
         // ToDo: Optimization Idea: Maybe we can save this websocket call and instead get the last message date from the conversation object in the post somehow?
         // send conversation with updated last message date to participants. This is necessary to show the unread messages badge in the client
-        conversationService.notifyAllConversationMembersAboutNewMessage(conversation, author);
+
+        notificationRecipients = notificationRecipients.stream().filter(user -> !Objects.equals(user.getId(), author.getId())).collect(Collectors.toSet());
+        // TODO: why do we need notification 2 and 3? we should definitely re-work this!
+        // Websocket notification 2
+        conversationService.notifyAllConversationMembersAboutNewMessage(course, conversation, notificationRecipients);
+
+        // creation of message posts should not trigger entity creation alert
+        // Websocket notification 3
+        conversationNotificationService.notifyAboutNewMessage(createdMessage, notificationRecipients, course);
```

Comment:
This are 3 different topics that are subscribed in different time: 

- notification 1 is when a conversation is active and we send the new messages to display.
- notification 2 is when metis service is setup and the users are under messages tab (all conversations are listed). This one is used when a user is added or removed from the conversation or a conversation setting is changed, this way we can update the client 
- notification 3 as you mentioned is the global notification.

Reply:
we have to improve this, but this is out of scope for this PR. I suggest we have a meeting with the involved developers and brainstorm about the best solution to avoid 3 notifications here